ql/src/java/org/apache/hadoop/hive/ql/exec/tez/KillMoveTriggerActionHandler.java

```diff
@@ -47,8 +47,10 @@ public void applyAction(final Map<WmTezSession, Trigger> queriesViolated) {
           break;
         case MOVE_TO_POOL:
           String destPoolName = entry.getValue().getAction().getPoolName();
-          Future<Boolean> moveFuture = wm.applyMoveSessionAsync(wmTezSession, destPoolName);
-          moveFutures.put(wmTezSession, moveFuture);
+          if (!wmTezSession.isDelayedMove()) {
+            Future<Boolean> moveFuture = wm.applyMoveSessionAsync(wmTezSession, destPoolName);
```

Comment:
I dont think we need to because any query being completed/killed in the destination pool will create a return/kill event. This  in turn will wake up the master thread which will retry the delayed move in the same iteration of the master thread loop. So the existing delayed moves will be processed earlier than any subsequent move events.